R/lint.R

```diff
@@ -42,38 +42,24 @@
 lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings = TRUE, text = NULL) {
   # TODO(next release after 3.0.0): remove this deprecated workaround
   dots <- list(...)
-  if (length(dots) > 0L && is.logical(dots[[1L]]) && !nzchar(names2(dots)[1L])) {
+  if (has_positional_logical(dots)) {
     warning(
       "'cache' is no longer available as a positional argument; please supply 'cache' as a named argument instead. ",
       "This warning will be upgraded to an error in the next release."
     )
     cache <- dots[[1L]]
     dots <- dots[-1L]
   }
-  if (is.null(text)) {
-    inline_data <- rex::re_matches(filename, rex::rex(newline))
-    if (inline_data) {
-      text <- gsub("\n$", "", filename)
-      lines <- strsplit(text, "\n", fixed = TRUE)[[1L]]
-      filename <- NULL
-    } else {
-      lines <- read_lines(filename)
-    }
-  } else {
-    inline_data <- TRUE
-    if (length(text) > 1L) {
-      text <- paste(text, collapse = "\n")
-    }
-    lines <- strsplit(text, "\n", fixed = TRUE)[[1L]]
-  }
 
-  no_filename <- missing(filename) || is.null(filename)
+  needs_tempfile <- missing(filename) || rex::re_matches(filename, rex::rex(newline))
```

Comment:
@AshesITR I've been starting at this a while for #2135... why are `needs_tempfile` and `inline_data` different?

Comparing:

```r
needs_tempfile <=>                   missing(filename) || rex::re_matches(filename, rex::rex(newline))
inline_data    <=> !is.null(text) || missing(filename) || rex::re_matches(filename, rex::rex(newline))
```

But don't we need `text=` present when `missing(filename)`?

Reply:
IIRC there was something with a filename being present and content passed as well for linter integrations that lint an unsaved buffer version of a file. But I'm not sure.

Reply:
Hmm, in that case, I will table #2135 until after release. Once we release, we can try disabling this/combining the conditions and see if it affects anything.